src/python/grpcio_tests/tests_aio/unit/call_test.py

```diff
@@ -124,24 +126,48 @@ class TestUnaryUnaryCall(AioTestBase):
 
             self.assertFalse(call.cancelled())
 
-            # TODO(https://github.com/grpc/grpc/issues/20869) remove sleep.
-            # Force the loop to execute the RPC task.
-            await asyncio.sleep(0)
-
             self.assertTrue(call.cancel())
             self.assertFalse(call.cancel())
 
-            with self.assertRaises(asyncio.CancelledError) as exception_context:
+            with self.assertRaises(grpc.RpcError) as exception_context:
                 await call
 
+            # The info in the RpcError should match the info in Call object.
+            rpc_error = exception_context.exception
+            self.assertEqual(rpc_error.code(), await call.code())
+            self.assertEqual(rpc_error.details(), await call.details())
+            self.assertEqual(rpc_error.trailing_metadata(), await
+                             call.trailing_metadata())
+            self.assertEqual(rpc_error.debug_error_string(), await
+                             call.debug_error_string())
+
             self.assertTrue(call.cancelled())
             self.assertEqual(await call.code(), grpc.StatusCode.CANCELLED)
             self.assertEqual(await call.details(),
                              'Locally cancelled by application!')
 
-            # NOTE(lidiz) The CancelledError is almost always re-created,
-            # so we might not want to use it to transmit data.
-            # https://github.com/python/cpython/blob/edad4d89e357c92f70c0324b937845d652b20afd/Lib/asyncio/tasks.py#L785
+    async def test_cancel_unary_unary_in_task(self):
+        async with aio.insecure_channel(self._server_target) as channel:
```

Comment:
I'm worried that this behavior could be surprising. I don't know of any equivalent credential in other projects, so it's unlikely that people would be familiar with this behavior. If we decided to go down this route, we would need to make the error message very clear when a connection is failed because you're trying to connect to a remote host with a `LocalCredentials`. And it would have to be a Python-level error, not a core-level debug log.